### PR TITLE
Cargo clippy linter fixes

### PR DESCRIPTION
**Pull request by andrei-marinica**
Nothing much to say, fixed some linter issues, not all.

It is not definitive, will need to be continued in the future.

### CODE REVIEW

File: elrond-wasm-debug/src/blockchain_mock.rs [hunk 1]
```diff
@@ -144,7 +158,7 @@ impl BlockchainMock {
 	fn get_new_address(&self, creator_address: Address, creator_nonce: u64) -> Option<Address> {
 		self.new_addresses
 			.get(&(creator_address, creator_nonce))
-			.map(|addr_ref| addr_ref.clone())
+			.cloned()
```
Comment (dorin-iancu) [inline elrond-wasm-debug/src/blockchain_mock.rs:161]: Typo or intentional? Shouldn't it be .clone() instead of .cloned()?

**Review by dorin-iancu — DISMISSED** [review]

File: elrond-wasm-debug/src/blockchain_mock.rs (hunk 1 above)
Comment (andrei-marinica) [reply elrond-wasm-debug/src/blockchain_mock.rs:161]: .cloned() is the iterator version

Comment (andrei-marinica) [reply elrond-wasm-debug/src/blockchain_mock.rs:161]: it means "clone each item from the iterator"

Comment (andrei-marinica) [reply elrond-wasm-debug/src/blockchain_mock.rs:161]: clippy proposed this

**Review by dorin-iancu — APPROVED** [review]

**Pull request merged**